packages/smart-contracts/test/contracts/ERC20SingleRequestProxy.test.ts

```diff
@@ -0,0 +1,152 @@
+    erc20SingleRequestProxy: ERC20SingleRequestProxy,
+    erc20FeeProxy: ERC20FeeProxy;
+
+  const paymentReference: BytesLike = '0xd0bc835c22f49e7e';
+  const feeAmount: BN = BN.from(10).mul(BASE_DECIMAL);
+
+  before(async function () {
+    [deployer, user1, user2, feeRecipient] = await ethers.getSigners();
+    user1Addr = await user1.getAddress();
+    user2Addr = await user2.getAddress();
+    feeRecipientAddr = await feeRecipient.getAddress();
+  });
+
+  beforeEach(async function () {
+    const deployerAddr = await deployer.getAddress();
+    testToken = await new TestToken__factory(deployer).deploy(deployerAddr);
+    await testToken.mint(deployerAddr, BN.from(1000000).mul(BASE_DECIMAL));
+
+    erc20FeeProxy = await new ERC20FeeProxy__factory(deployer).deploy();
+    erc20SingleRequestProxy = await new ERC20SingleRequestProxy__factory(deployer).deploy(
+      user2Addr,
+      testToken.address,
+      feeRecipientAddr,
+      feeAmount,
+      paymentReference,
+      erc20FeeProxy.address,
+    );
+
+    await testToken.transfer(user1Addr, BN.from(10000).mul(BASE_DECIMAL));
+    await testToken
+      .connect(user1)
+      .approve(erc20SingleRequestProxy.address, ethers.constants.MaxUint256);
+  });
+
+  it('should be deployed', async () => {
+    expect(erc20SingleRequestProxy.address).to.not.equal(ethers.constants.AddressZero);
+  });
+
+  it('should set the correct initial values', async () => {
+    expect(await erc20SingleRequestProxy.payee()).to.equal(user2Addr);
+    expect(await erc20SingleRequestProxy.tokenAddress()).to.equal(testToken.address);
+    expect(await erc20SingleRequestProxy.feeAddress()).to.equal(feeRecipientAddr);
+    expect(await erc20SingleRequestProxy.feeAmount()).to.equal(feeAmount);
+    expect(await erc20SingleRequestProxy.paymentReference()).to.equal(paymentReference);
+    expect(await erc20SingleRequestProxy.erc20FeeProxy()).to.equal(erc20FeeProxy.address);
+  });
+
+  it('should process a payment correctly', async () => {
+    const paymentAmount = BN.from(100).mul(BASE_DECIMAL);
+    const totalAmount = paymentAmount.add(feeAmount);
+
+    await testToken.connect(user1).transfer(erc20SingleRequestProxy.address, totalAmount);
+
+    const erc20SingleRequestProxyBalanceBefore = await testToken.balanceOf(
+      erc20SingleRequestProxy.address,
+    );
+    expect(erc20SingleRequestProxyBalanceBefore).to.equal(totalAmount);
+
+    await user1.sendTransaction({
+      to: erc20SingleRequestProxy.address,
+      value: 0,
+    });
+
+    const erc20SingleRequestProxyBalanceAfter = await testToken.balanceOf(
+      erc20SingleRequestProxy.address,
+    );
+    const user2BalanceAfter = await testToken.balanceOf(user2Addr);
+    const feeRecipientBalanceAfter = await testToken.balanceOf(feeRecipientAddr);
+
+    expect(erc20SingleRequestProxyBalanceAfter).to.equal(0);
+    expect(user2BalanceAfter).to.equal(paymentAmount);
+    expect(feeRecipientBalanceAfter).to.equal(feeAmount);
```

Comment:
Can we make this test check that the `paymentReference` event was emitted, like the analogous EthereumSingleRequestProxy test: `should process a payment correctly and emit event`